dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/listener/EnrollmentSMSListener.java

```diff
@@ -154,11 +158,16 @@ protected SMSResponse postProcess( IncomingSms sms, SMSSubmission submission )
 
         TrackedEntityInstance tei = teiService.getTrackedEntityInstance( teiUID.uid );
 
+        // return response if an ACTIVE enrollment exists
+        if ( enrollmentExists( tei, program ) )
```

Comment:
This SMSSubmission could be used to create and edit existing enrollments (for example, to modify enrollment status or enrollment date). I feel this check excludes the edition of existing enrollments.

Also, in case this submission is used to modify the TEI attribute values in an existing enrollment, the values will be correctly modified (just a few lines above) but the response will be a kind of error instead of success.